### PR TITLE
Add schema bsed on diagram

### PR DESCRIPTION
**Pull request by Nowembabazi**
##In this milestone we achieved the following;
-CREATED A SCHEMA BASED ON THE DIAGRAM.
-Joined tables from many-to-many relationships.

### CODE REVIEW

File: schema_based_on_diagram [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+CREATE TABLE treatments(
+id INTEGER NOT NULL,
+type VARCHAR(150) NOT NULL,
+name VARCHAR(150) NOT NULL,
+PRIMARY KEY(id)
+);
+
+CREATE TABLE medical_histories_treatments(
+medical_histories_id INTEGER NOT NULL,
+treatment_id INTEGER NOT NULL,
```
Comment (t-yanick) [inline schema_based_on_diagram]: - [ ] In this join table for many-to-many relationship, all FOREIGN KEYS are set by using `refrences table_name(id)` in column definition. This implies that for line 35 and 36, you should have something like below:
- `medical_history_id int refrences medical_histories(id),`
- `treatment_id int refrences treatments(id),`
Please also verify the column name, if it should be `medical_histories_id `or `medical_history_id`

File: schema_based_on_diagram [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+PRIMARY KEY(medical_histories_id, treatment_id)
```
Comment (t-yanick) [inline schema_based_on_diagram]: - [ ] The join table has FOREIGN KEYS which are the PRIMARY KEYS of the `medical_histories` and `treatments` table. I don't think it is necessary to mention again PRIMARY KEY as in line 37 as there are already references to the original tables

File: schema_based_on_diagram [hunk 3]
```diff
@@ -0,0 +1,50 @@
+CREATE TABLE patients(
+    id INTEGER NOT NULL,
+    name VARCHAR(150) NOT NULL,
+    date_of_birth DATE NOT NULL,
+    PRIMARY KEY(id)
+);
+
```
Comment (t-yanick) [inline schema_based_on_diagram:7]: - [ ] In your project folder, I cannot find the correct diagram. Please kindly update with the diagram in the screenshot below.
image
- Below is what I can see in your project folder
image

File: schema_based_on_diagram [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+id INTEGER NOT NULL,
+unit_price DECIMAL(8, 2) NOT NULL,
+quantity INTEGER NOT NULL,
+total_price DECIMAL(8, 2) NOT NULL,
+invoice_id INTEGER NOT NULL,
+treatment_id INTEGER NOT NULL,
+PRIMARY KEY(id),
+CONSTRAINT invoice_items_invoice_id_foreign FOREIGN KEY(invoice_id) REFERENCES invoices(id),
+CONSTRAINT invoice_items_treatment_id_foreign FOREIGN KEY(treatment_id) REFERENCES treatments(id)
+);
```
Comment (t-yanick) [inline schema_based_on_diagram]: - [ ] One of the project requirements as you can see in the screenshot below is for all FOREIGN KEYS to have indexes created, but you don't have any indexes created for your FOREIGN KEYS. Kindly fix this.
image

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @Nowembabazi and @Ngufor-emmanuella , 

### Good job so far!

### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram (hunk 2 above)
Comment (Abenezer-Tilahun) [reply schema_based_on_diagram]: Fixed.

File: schema_based_on_diagram (hunk 4 above)
Comment (Abenezer-Tilahun) [reply schema_based_on_diagram]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Nowembabazi 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ngufor-emmanuella** [comment]
thank you

**Pull request merged**